### PR TITLE
Feature validate user account

### PR DESCRIPTION
**Pull request by phawazzzy**
this feature creates an endpoint to receive the URL clicked from the email received by the user
it validates  if the token has not expired
checks if the  user has an account on the database,
then activate the user's account

### CODE REVIEW

File: api/controllers/user.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+exports.signup = async (req, res) => {
+  try {
+    // get values from body
+    const {
+      username, email, password, confirmPassword
+    } = req.body;
+    // checks if password and confirmPasword matche
+    if (password !== confirmPassword) {
+      return res.status(403).json({ response: "confirm password not matched with password" });
```
Comment (AminuIsrael) [inline api/controllers/user.js:20]: Typo

File: api/controllers/user.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    // hash password
+    const hash = await hashPassword(confirmPassword);
+    const token = await generateToken({ email });
+    console.log(token);
```
Comment (AminuIsrael) [inline api/controllers/user.js]: You're not suppose to add console.log() to the code na

**Review by AminuIsrael — CHANGES_REQUESTED** [review]

File: api/controllers/user.js (hunk 1 above)
Comment (phawazzzy) [reply api/controllers/user.js:20]: Where

File: api/controllers/user.js (hunk 2 above)
Comment (phawazzzy) [reply api/controllers/user.js]: How did that escaped eslint and codacy check... Will fix now

File: api/controllers/user.js (hunk 1 above)
Comment (AminuIsrael) [reply api/controllers/user.js:20]: Okay

File: api/controllers/user.js (hunk 2 above)
Comment (AminuIsrael) [reply api/controllers/user.js]: Okay

File: api/controllers/user.js (hunk 1 above)
Comment (AminuIsrael) [reply api/controllers/user.js:20]: You can use 'Invalid Password' No too much story

File: api/controllers/user.js (hunk 2 above)
Comment (phawazzzy) [reply api/controllers/user.js]: it has been fixed ..this is flagged outdated ..it must be one of the previous commits that have been fixed  in the new commits

**Comment by adefemi171** [comment]
@phawazzzy why are you declaring `const { token } = req.query` why not `const token = req.query`

**Comment by phawazzzy** [comment]
> @phawazzzy why are you declaring `const { token } = req.query` why not `const token = req.query`

@adefemi171 it is supposed to be written as `const token  = req.query.token`, but the eslint rules prefer object destructuring that is why it is written like that.
`req.query` returns an object of all query parameters in that route. so to get the token query parameter

check the express documentation here [req.query](https://expressjs.com/en/api.html#req.query)

**Pull request merged**